c7n/resources/asg.py

```diff
@@ -492,6 +492,49 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('failing')
+class FailingFilter(Filter):
+    schema = type_schema(
+        'failing', count={'type': 'number'})
+
+    def asg_failed_launch(self, activities={}, token=None):
+        limit_date = datetime.now(tz=tzutc()) - timedelta(days=1)
+        client = local_session(
+            self.manager.session_factory).client('autoscaling')
+        if token:
+            results = client.describe_scaling_activities(NextToken=token)
```

Comment:
use the client method get_paginator('describe_scaling_activities') to get a paginator, compare the results on each page to the comparison date and stop the pagination when the target date is reached.